Source/gmenu.cpp

```diff
@@ -109,7 +109,7 @@ void gmenu_call_proc(TMenuItem *pItem, void (*gmFunc)(TMenuItem *))
 			sgCurrentMenuIdx++;
 		}
 	}
-	sgpCurrItem = &sgpCurrentMenu[sgCurrentMenuIdx - 1];
+	sgpCurrItem = sgCurrentMenuIdx > 0 ? &sgpCurrentMenu[sgCurrentMenuIdx - 1] : nullptr;

```

Comment:
I wonder whether the intent here was `sgpCurrentMenu[0]` or `nullptr`.

Reply:
We should probably add a BUGFIX comment for this upstream

Reply:
Sent https://github.com/diasurgical/devilution/pull/1799

Reply:
Looks like this can only happen if you create a menu with no items, in such a case it would still be wrong to select item 0 so null looks more correct. I didn't look into how this would play out later in the code, but since this is an edge case that would only happen if someone were modding the game and created an empty menu I don't think it matters much.

Reply:
@AJenbo it happens when starting the game and opening the menu, that's how I knew about this (UBSAN complained)

Reply:
Oh, I see now, this is called with NULL, NULL in order to clear the menu. Defiantly a bug and definitely should be NULL like you did in this patch :)